sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/main/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistrySchemaCache.java

```diff
@@ -0,0 +1,228 @@
+            serviceCall = this.schemaRegistryClient
+                .registerSchema(schemaGroup, schemaFullName, schemaDefinition, SchemaFormat.JSON);
+        } else {
+            serviceCall = this.schemaRegistryClient.getSchemaProperties(
+                schemaGroup, schemaFullName, schemaDefinition, SchemaFormat.JSON);
+        }
+
+        return serviceCall.map(properties -> {
+            final String schemaId = properties.getId();
+
+            synchronized (lock) {
+                cache.put(schemaId, schemaDefinition);
+                logCacheStatus();
+            }
+
+            return schemaId;
+        });
+    }
+
+    Mono<String> getSchema(String schemaId) {
+        synchronized (lock) {
+            final String existing = cache.get(schemaId);
+
+            if (existing != null) {
+                return Mono.just(existing);
+            }
+        }
+
+        return schemaRegistryClient.getSchema(schemaId)
+            .handle((registryObject, sink) -> {
+                final String schemaString = registryObject.getDefinition();
+
+                synchronized (lock) {
+                    cache.put(schemaId, schemaString);
+                    logCacheStatus();
+                }
+
+                sink.next(schemaString);
+            });
+    }
+
+    /**
+     * Gets number of cached schemas.
+     *
+     * @return Number of cached schemas.
+     */
+    int getSize() {
+        synchronized (lock) {
+            return cache.size();
+        }
+    }
+
+    /**
+     * Gets the length of schemas stored in cache.
+     *
+     * @return The length of schemas stored in cache.
+     */
+    int getTotalLength() {
+        synchronized (lock) {
+            return cache.getTotalLength();
+        }
+    }
+
+    /**
+     * Logs the cache status if log level verbose is enabled. Otherwise, no-op.
+     */
+    private void logCacheStatus() {
+        if (!logger.canLogAtLevel(LogLevel.VERBOSE)) {
+            return;
+        }
+
+        final int size = cache.size();
+        final int length = cache.getTotalLength();
+
+        logger.atVerbose()
+            .addKeyValue(SIZE_KEY, size)
+            .addKeyValue(TOTAL_LENGTH_KEY, length)
+            .log("Cache entry added or updated. Total number of entries: {}; Total schema length: {}",
+                size, length);
+    }
+
+    /**
+     * Simple LRU cache. Accesses to cache are synchronized via the outer class lock.
+     * TODO: When https://github.com/Azure/azure-sdk-for-java/pull/27408/ is merged, take a look at replacing.
+     */
+    private static final class SchemaCache extends LinkedHashMap<String, String> {
+        private static final long serialVersionUID = -1L;
+
+        private final int capacity;
+        /**
+         * Map of JSON schema definitions to schema ids.
+         * Key: JSON schema definition
+         * Value: Schema Id
+         */
+        private final HashMap<String, String> schemaToIdCache = new HashMap<>();
```

Comment:
just curious why we compose this HashMap schemaToIdCache, when the SchemaCache is already a HashMap

Reply:
There are two mapping directions.
A) Key: SchemaId, Value: Schema Definition
B) Key: Schema Definition, Value: SchemaId